sdk/azurestackhci/Azure.ResourceManager.Hci/src/Generated/OfferData.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.ResourceManager.Hci.Models;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.Hci
+{
+    /// <summary> A class representing the Offer data model. </summary>
+    public partial class OfferData : ResourceData
+    {
+        /// <summary> Initializes a new instance of OfferData. </summary>
+        public OfferData()
+        {
+            SkuMappings = new ChangeTrackingList<SkuMappings>();
+        }
+
+        /// <summary> Initializes a new instance of OfferData. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="provisioningState"> Provisioning State. </param>
+        /// <param name="publisherId"> Identifier of the Publisher for the offer. </param>
+        /// <param name="content"> JSON serialized catalog content of the offer. </param>
+        /// <param name="contentVersion"> The API version of the catalog service used to serve the catalog content. </param>
+        /// <param name="skuMappings"> Array of SKU mappings. </param>
+        internal OfferData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, string provisioningState, string publisherId, string content, string contentVersion, IList<SkuMappings> skuMappings) : base(id, name, resourceType, systemData)
+        {
+            ProvisioningState = provisioningState;
+            PublisherId = publisherId;
+            Content = content;
+            ContentVersion = contentVersion;
+            SkuMappings = skuMappings;
+        }
+
+        /// <summary> Provisioning State. </summary>
+        public string ProvisioningState { get; }
```

Comment:
Should this be of type [HciProvisioningState.cs](https://github.com/Azure/azure-sdk-for-net/pull/33589/files#diff-226260d0b2f6f1106c429b82da43246a2ce629877d53bba0f587db4a34c9705c)?

Reply:
Changed